portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +25,130 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    private List<String> countries;
+    private List<String> capitals;
+
+    @Override
+    public void init() {
+        countries = new ArrayList();
```

Comment:
Structuring your data this way makes it very hard to maintain / error prone, because it's critical to update matching entries in two independent lists.

This would be much clearer and maintainable with a more complex data structure.

```
public static class Country {
  private final String name, capital;
  public Country(String name, String capital) { 
  ...
}

countries.add(new Country("Germany", "Berlin"));
```
You can then be sure that both values are always handled/updated together, and that there's no risk that the lists will have e.g. different lengths or that future edits are made in the wrong position.


A worse (but better than two lists) approach is to use a `Map<String, String>` countries to represent the name/capital as a single entry. You can iterate over the `.entrySet()` or access a specific countries capital directly.